src/main/java/org/cryptomator/common/ErrorCodeGenerator.java

```diff
@@ -0,0 +1,53 @@
+package org.cryptomator.common;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.Locale;
+
+@Singleton
+public class ErrorCodeGenerator {
+
+
+	private final static int A_PRIME = Integer.MAX_VALUE;
+
+	private final static int LATEST_FRAME = 1;
+	private final static int ALL_FRAMES = Integer.MAX_VALUE;
+
+	@Inject
+	ErrorCodeGenerator() { /* NO-OP */ }
+
+	public String of(Throwable e) {
+		Preconditions.checkNotNull(e);
+		return format(traceCode(rootCause(e), LATEST_FRAME)) + ':' + format(traceCode(e, ALL_FRAMES));
+	}
+
+	private Throwable rootCause(Throwable e) {
+		if (e.getCause() == null) {
+			return e;
+		}
+		return rootCause(e.getCause());
+	}
+
+	private String format(int value) {
+		//Cut off highest 12 bits (only leave 20 least significant bits) and XOR rest with cutoff
+		value = (value & 0xfffff) ^ (value >>> 20);
+		return Strings.padStart(Integer.toString(value, 32).toUpperCase(Locale.ROOT), 4, '0');
+	}
+
+	private int traceCode(Throwable e, int frameCount) {
+		int result = 0x6c528c4a;
```

Comment:
Can we please replace this with a nothing-up-my-sleeve number? Each time I read this, I feel the urge to understand _why_ this number has been chosen.

Also, make it a constant, e.g. `private static final int SEED = 0x12345678`.

Reply:
solved in 23ff938 and 3ba7450